consensus/hotstuff/signature/weighted_signature_aggregator.go

```diff
@@ -59,74 +60,54 @@ func NewWeightedSignatureAggregator(
 		}
 	}
 
-	// build a low level crypto aggregator
+	// instantiate low-level crypto aggregator, which works based on signer indices instead of nodeIDs
 	agg, err := signature.NewSignatureAggregatorSameMessage(message, dsTag, pks)
 	if err != nil {
-		return nil, fmt.Errorf("new signature aggregator failed: %w", err)
-	}
-
-	// build the weighted aggregator
-	weightedAgg := &WeightedSignatureAggregator{
-		aggregator:   agg,
-		ids:          ids,
-		idToInfo:     idToInfo,
-		collectedIDs: make(map[flow.Identifier]struct{}),
+		return nil, fmt.Errorf("instantiating index-based signature aggregator failed: %w", err)
 	}
 
-	return weightedAgg, nil
+	return &WeightedSignatureAggregator{
+		aggregator: agg,
+		ids:        ids,
+		idToInfo:   idToInfo,
+	}, nil
 }
 
-// Verify verifies the signature under the stored public and message.
-//
-// The function errors:
-//  - engine.InvalidInputError if signerID is invalid (not a consensus participant)
-//  - module/signature.ErrInvalidFormat if signerID is valid but signature is cryptographically invalid
-//  - random error if the execution failed
+// Verify verifies the signature under the stored public keys and message.
+// Error returns:
+//  - model.InvalidSignerError if signerID is invalid (not a consensus participant)
+//  - model.ErrInvalidSignature if signerID is valid but signature is cryptographically invalid
+//  - generic error in case of unexpected runtime failures
```

Comment:
shall we use exception instead of generic error? I believe we use exception in most of the cases

Reply:
I often write:
```golang
// Expected errors during normal operations:
//  - model.InvalidSignerError if signerID is invalid (not a consensus participant)
//  - model.ErrInvalidSignature if signerID is valid but signature is cryptographically invalid
```
(removes one extra line; thereby avoiding the overhead of always explicitly mentioning generic errors)